### PR TITLE
Financial Terms: Fixing / Expanding Asset Terms, Fixing Loan Term Date Calculations, Adding Semiannual Term

### PR DESCRIPTION
**Pull request by Windchild292**
This is a mixture of a bugfix and enhancement, and was raised originally on Discord.

Bugs:
1. Fixing the disconnect between allowing Quarterly and implementing Monthly terms for assets.
2. Fixing multiple loan date calculation issues

Enhancements:
1. Adding Semiannual Financial term
2. Enabling Biweekly, Monthy, and Semiannual Financial Terms for Assets

(http://5ko.free.fr/en/year.php?y=3067&s= is great for testing)

### CODE REVIEW

**Comment by Windchild292** [comment]
This is ready for review.

File: MekHQ/src/mekhq/campaign/finances/enums/FinancialTerm.java [hunk 1]
```diff
@@ -68,33 +70,94 @@ public boolean isQuarterly() {
...
+    public LocalDate nextValidDate(final LocalDate yesterday, final LocalDate today) {
         switch (this) {
             case BIWEEKLY:
-                return origin.with(TemporalAdjusters.nextOrSame(DayOfWeek.MONDAY)).plusWeeks(2);
+                // First, find the first day of the current week
+                final LocalDate date = today.with(WeekFields.ISO.dayOfWeek(), 1);
+                // Second, add two weeks to the date if this is an even week, or three if it is not
+                return date.plusWeeks(
+                        ((date.get(WeekFields.ISO.weekOfYear()) % 2) == 0)
+                                ? 2 : 3);
```
Comment (rjhancock) [inline MekHQ/src/mekhq/campaign/finances/enums/FinancialTerm.java:105]: BiWeekly is every 2 weeks. BiMonthly is twice a month (or every 2 months).

So shouldn't this just be every 2 weeks?

Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/finances/enums/FinancialTerm.java:105]: We use a built-in grace period that lines everything up to the first day of the financial setup (Monday, First of the Month, First of the Year, etc). I didn't want to change the internal decision for that, so this actually handles that decision properly.

Comment (rjhancock) [reply MekHQ/src/mekhq/campaign/finances/enums/FinancialTerm.java:105]: So this is basically saying "if this is in the middle of a bi-weekly period, add a week" and locks the biweekly literally into every even week vs time based every two weeks?

Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/finances/enums/FinancialTerm.java:105]: Exactly, same way as it was meant to be handled previously but without the calculation bugs.

**Review by rjhancock — APPROVED** [review]

**Pull request merged**